src/main/java/games/strategy/debug/ConsoleHandler.java

```diff
@@ -0,0 +1,105 @@
+package games.strategy.debug;
+
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.logging.ErrorManager;
+import java.util.logging.Handler;
+import java.util.logging.Level;
+import java.util.logging.LogManager;
+import java.util.logging.LogRecord;
+import java.util.logging.SimpleFormatter;
+
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * A {@link Handler} that publishes log records of level {@link Level#WARNING} or {@link Level#SEVERE} to
+ * {@code System.err} and log records of all other levels to {@code System.out}.
+ *
+ * <p>
+ * <strong>Configuration:</string> This handler does not currently support configuration through the {@link LogManager}.
+ * It always uses the following default configuration:
+ * </p>
+ *
+ * <ul>
+ * <li>Level: {@code Level.ALL}</li>
+ * <li>Filter: No {@code Filter}</li>
+ * <li>Formatter: {@code java.util.logging.SimpleFormatter}</li>
+ * <li>Encoding: default platform encoding</li>
+ * </ul>
+ */
+public final class ConsoleHandler extends Handler {
+  private final Supplier<PrintStream> errSupplier;
+  private final Supplier<PrintStream> outSupplier;
+
+  public ConsoleHandler() {
+    this(() -> System.out, () -> System.err);
+  }
+
+  @VisibleForTesting
+  ConsoleHandler(final Supplier<PrintStream> outSupplier, final Supplier<PrintStream> errSupplier) {
+    this.errSupplier = errSupplier;
+    this.outSupplier = outSupplier;
+    configure();
+  }
+
+  private void configure() {
+    setLevel(Level.ALL);
+    setFilter(null);
```

Comment:
Is out-of-scope for this PR, but what about adding another option in the properties file which filters logs which levels are below info etc.
If we'd log every detail, we could adjust this level to get more or less information, which makes debugging easier.
In general we need a unified logging system.
Currently those loggers for the server side stuff and the ClientLogger classes are been used for logging

Reply:
> what about adding another option in the properties file which filters logs which levels are below info etc.

The default configuration does filter logs below `INFO`.  In the Java logging framework, there's two log levels at play:

1. The log level associated with the `Logger` that publishes the log.
1. The log level associated with the `Handler` that receives the log.

The default configuration uses a log level of `INFO` for all `Logger`s  and a log level of `ALL` for the `ConsoleHandler`.  The intersection of those two yields an overall log level of `INFO`.

> we could adjust this level to get more or less information

There are a few ways to do that.  In a dev environment, you can simply edit the `logging.properties` file.  In a non-dev environment, you can specify your own `.properties` file to override the default configuration by passing `-Djava.util.logging.config.file=/path/to/logging.properties` as a JVM argument when starting TripleA.

> Currently those loggers for the server side stuff and the ClientLogger classes are been used for logging

Yeah, now that logs at different levels are split between stdout and stderr, usage of the `ClientLogger` class can probably be replaced with Java `Logger`s.  `logError()` calls would most likely correspond to the `SEVERE` level.  `logQuietly()` calls would correspond to no higher than `INFO` level, but we'd probably want to take the opportunity to turn down the level even more on certain logs if possible.

Using the Java log framework everywhere would also give us the ability to tee the complete logs to a file simply by adding a `FileHandler` to the configuration.  When trying to debug a problem, we could provide users with a custom `logging.properties` file that would do this, and then just ask them to attach the resulting log file to the issue.

Reply:
> Using the Java log framework everywhere would also give us the ability to tee the complete logs to a file simply by adding a FileHandler to the configuration. When trying to debug a problem, we could provide users with a custom logging.properties file that would do this, and then just ask them to attach the resulting log file to the issue.

Really good idea! This would open so much possibilities! (We could maybe also implement an automatic logging framework which uploads stacktraces to a server so we can see if something goes wrong for example, currently not needed though ^^), but we should probably make sure logs are regularily cleared so the file size doesn't constantly increase until the drive is full.

The protocol fix will be delayed by an hour btw. RL comes first ^^

Reply:
> we should probably make sure logs are regularily cleared so the file size doesn't constantly increase until the drive is full.

`java.util.logging.FileHandler` supports this out-of-the-box. :smile:  It also supports file rotation across N files.